test-data/unit/check-classes.test

```diff
@@ -6512,3 +6512,28 @@ def access_after_declaration(self) -> None:
 
     reveal_type(x)  # N: Revealed type is 'builtins.int'
     x = x + 1
+
+[case testIsSubClassNarrowDownTypesOfTypeVariables]
+from typing import TypeVar, Generic
+
+TypeT = TypeVar("TypeT", bound=type)
+
+class IntBase:
+    field: int = 42
+
+class IntFoo(Generic[TypeT]):
+    def method(self, other: TypeT) -> int:
+        if issubclass(other, IntBase):
+            return other.field
```

Comment:
Relying only on return type to test such cases is not safe, because the attribute may be `Any`. I would additionally add a `reveal_type()` above.